lib/puppet/provider/mysql.rb

```diff
@@ -38,11 +38,35 @@ class Puppet::Provider::Mysql < Puppet::Provider
   ].join(':')
 
   # rubocop:disable Style/HashSyntax
-  commands :mysql_raw  => 'mysql'
-  commands :mysqld     => 'mysqld'
-  commands :mysqladmin => 'mysqladmin'
+  commands :mysql_client     => 'mysql'
+  commands :mariadb_client   => 'mariadb'
```

Comment:
You should probably use `optional_commands` to be explicit that they're not expected to be present (or the `has_command` definition together with `is_optional`).